src/client/consumer.ts

```diff
@@ -19,9 +22,11 @@ export interface RecordReceivedEvent {
 export interface ConsumedRecord {
     topic: string;
     value: string | Buffer | null;
+    deserializedValue?: SerializationdResult | null;
     offset?: string;
     partition?: number;
     key?: string | Buffer;
+    deserializedKey?: SerializationdResult | null;
```

Comment:
why do we need to keep both?

Reply:
It was because when record was generated,it gerenates deserializedKey with a null value and the consumer view checked this value. I push a new code version. Now I generate deserializedKey for record only if the result of the deserializer return != null. Andin consumer view I can check if message.deserializedKey is undefined to know if I must display the deserialized key or the real key.

Reply:
fixed

Reply:
There's no reason why the message should keep both values. It should simply hold the values (deserialized or raw)

Reply:
Ok I understand more what you mean. Good catch! Now I have just `key `and `value` field which clean the code.Thanks for your comment.